index.js

```diff
@@ -31,13 +32,16 @@ function addToDoItem(event) {
 }
 
 function noteChecker (e) {
-    if (e.target.checked) {
-        
-        document.querySelector(`#item-${numberOfItems} > h2`).classList.add("checked");
-    }
-    else{
-        document.querySelector(`#item-${numberOfItems} > h2`).classList.remove("checked");
-    }
+        let note = e.composedPath()[1];

```

Comment:
Had a look at composed path after you mentioned it, and does look like the better option than just plain path, nice find!